### ISSUE TITLE
Corrupted bitstream with speeds 2 to 5, quantizer <= 30

### ISSUE THREAD

**Issue by SmilingWolf** [xiph/rav1e#1649]
rav1e commit: 07ca598

System:
CPU: i7-4770 (with AVX2)
OS: Win7 64bit

Clip: https://mega.nz/#!5tp2nIgL!mFJnlgtY6r2O87_OHo3eItfMr9h9j9jDeoNtQ4lb3S0
Encoded file: [rav1e.cq30sp2.bugRepro.shorter.zip](https://github.com/xiph/rav1e/files/3597709/rav1e.cq30sp2.bugRepro.shorter.zip)

sha1sums:
```
781074334db079c9dd3c5d63ddc859b13d3a67be *rav1e.cq30sp2.bugRepro.shorter.y4m    (original clip)
823d6e103f2cbd24e62f140f4848d7320dd05c7d *rav1e.cq30sp2.bugRepro.shorter.mkv    (ffv1 compressed original clip)
576ebb4b3897aa95ae03e2e6b2b2eb18b69f3328 *rav1e.cq30sp2.bugRepro.shorter.ivf    (encoded clip)
359656d8c8e618867ead43e8f6d6e7cc40544dc9 *rav1e.cq30sp2.bugRepro.shorter.zip    (encoded clip archive)
```

Cmdlines used:
```
# rav1e -0
rav1e 0.1.0 (20190430-473-g07ca598)
# rav1e -o rav1e.cq30sp2.bugRepro.shorter.ivf --quantizer 30 -s 2 --tune Psnr rav1e.cq30sp2.bugRepro.shorter.y4m
...
# aomdec -o /dev/null rav1e.cq30sp2.bugRepro.shorter.ivf
Warning: Failed to decode frame 17: Corrupt frame detected
Warning: Additional information: Corrupted segment_ids
```

**Comment by SmilingWolf** [comment]
I have done more tests. So far, I've been able to generate corrupted bitstreams for speeds 2, 3, 4 and 5.
The condition seems to be `--quantizer <= 30`.
At speed 5, quantizer values of 28, 29 and 30 give corrupted bitstreams, 31 and up the bitstream is good.

**Comment by ycho** [comment]
@tdaede: So, you have confirmed that segment based q_idx become <=1 is the sole reason for this breakage, right?

**Comment by ycho** [comment]
> `--quantizer 10 --speed 5 --limit 1 --reconstruction rec.y4m broken.y4m -o test.ivf` on latest master with the above patch fails being decoded by aomdec, but can be decoded by dav1d, although with noticeable artifacts in the bottom row (absent from the reconstruction):
> ![image](https://user-images.githubusercontent.com/1794388/64827422-fd103380-d578-11e9-97ac-dec0b4cce69e.png)

As similar to the comment in: https://github.com/xiph/rav1e/issues/1719#issuecomment-539225461,
when this PR is filed, quant_rdo is only enabled for speed 0,1,2,3, thus "--quantizer 10" would be the reason that varies frame to frame by the rate controller.

**Comment by ycho** [comment]
> Reproduced with `--quantizer 30 --speed 5 --limit 2 broken.y4m -o test.ivf`. `--limit 1` works fine.

Reproducing should have used speed 3 or smaller, if the issue was with quant_rdo/
Since there was issue with speed 5 as above (https://github.com/xiph/rav1e/issues/1649#issuecomment-531042180) and now it was fixed by #1694, the issue does not seem to related to quant_rdo.

**Comment by ycho** [comment]
Oh...
"  /// When disabled, the segment ID is chosen heuristically.
  pub quantizer_rdo: bool,"
I missed whole this. So, seg_id still works even if quantizer_rdo is disabled.
Based on block importance?
